### PR TITLE
Forms Update

### PR DESCRIPTION
**Pull request by ericnakagawa**
Fixes [8052](https://github.com/facebook/react/issues/8052)
Fixes [8068](https://github.com/facebook/react/issues/8068)

* Removed .includes() and updated corresponding CodePen
* Added line-highlighting on first 2 examples

### CODE REVIEW

File: docs/docs/forms.md [hunk 1]
```diff
@@ -91,9 +93,9 @@ An **uncontrolled** component manages its own state.
   }
 ```
 
-If you wanted to listen to updates to the value, you could use the `onChange` event just like you can with controlled components.
+If you wanted to listen to updates to the value, you could use the `onChange` event just like you can with controlled components, however you would _not_ pass the value you saved to the component.
```
Comment (lacker) [inline docs/docs/forms.md]: Just a phrasing nitpick - can you break this up into two sentences like:

> If you wanted to listen to updates to the value, you could use the `onChange` event just like you can with controlled components. However, you would not pass the value you saved to the component.

File: docs/docs/forms.md [hunk 2]
```diff
@@ -376,38 +379,41 @@ ReactDOM.render(<Form />, document.getElementById('root'));
...
-    let val = event.target.value;
-    let checked = this.state.checked.slice(); // copy
-    if(checked.includes(val)) {
-      checked.splice(checked.indexOf(val), 1);
-    } else {
-      checked.push(val);
-    }
+    let value = event.target.value;
+    let checked = this.state.checked; // copy 
+    if(!checked[value]) { checked[value] = true; } else { checked[value] = false; }
```
Comment (lacker) [inline docs/docs/forms.md]: can you put a space after `if` and break this up into multiple lines so it matches the style of other parts?

File: docs/docs/forms.md [hunk 3]
```diff
@@ -376,38 +379,41 @@ ReactDOM.render(<Form />, document.getElementById('root'));
...
     this.setState({checked: checked})
   }
 
   handleSubmit(event) {
-    alert("Boxes checked are: '" + this.state.checked + "'");
+    alert("Boxes checked: " +
+      (this.state.checked.A?"A ":"") +
```
Comment (lacker) [inline docs/docs/forms.md]: can you put spaces around the `?` and `:` operators when you're ternarying

File: docs/docs/forms.md [hunk 4]
```diff
@@ -376,38 +379,41 @@ ReactDOM.render(<Form />, document.getElementById('root'));
...
+    let value = event.target.value;
+    let checked = this.state.checked; // copy 
+    if (!checked[value]) { checked[value] = true; } else { checked[value] = false; }
     this.setState({checked: checked})
   }
 
   handleSubmit(event) {
-    alert("Boxes checked are: '" + this.state.checked + "'");
+    alert("Boxes checked: " +
+      (this.state.checked.A ? "A ":"") +
```
Comment (lacker) [inline docs/docs/forms.md]: spaces around the `:` too, plz

**Comment by gaearon** [comment]
Thank you!

**Pull request merged**